### PR TITLE
GCS template loading + alert templates for slack

### PR DESCRIPTION
**Pull request by ajvb**

### CODE REVIEW

File: src/main/java/com/mozilla/secops/GcsUtil.java [hunk 1]
```diff
@@ -1,18 +1,47 @@
...
+    }
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+    // check if blob exists
+    Blob blob = storage.get(bid);
+    if (blob == null || !blob.exists()) {
+      return null;
+    }
+    try {
+      return storage.signUrl(BlobInfo.newBuilder(bid).build(), 10, TimeUnit.MINUTES);
+    } catch (IllegalStateException exc) {
```
Comment (ameihm0912) [inline src/main/java/com/mozilla/secops/GcsUtil.java]: Should these be propagated up?

File: src/main/java/com/mozilla/secops/alert/Alert.java [hunk 2]
```diff
@@ -263,22 +263,41 @@ public String getCategory() {
...
   /**
-   * Get template name
+   * Get email template name
    *
    * @return Freemarker template name with file extension or null if not set.
    */
   @JsonIgnore
-  public String getTemplateName() {
-    return getMetadataValue("template_name");
+  public String getEmailTemplateName() {
```
Comment (ameihm0912) [inline src/main/java/com/mozilla/secops/alert/Alert.java]: Will one always be applicable? Do we need some sort of postfix for the type of alert, e.g., direct, etc?

File: src/main/java/com/mozilla/secops/authprofile/AuthProfile.java [hunk 3]
```diff
@@ -251,6 +251,12 @@ private Alert createBaseAlert(Event e) {
         a.addMetadata("sourceaddress_country", "unknown");
       }
 
+      if (e.getNormalized().isOfType(Normalized.Type.AUTH)) {
+        a.addMetadata("auth_alert_type", "auth");
+      } else if (e.getNormalized().isOfType(Normalized.Type.AUTH_SESSION)) {
+        a.addMetadata("auth_alert_type", "auth");
```
Comment (ameihm0912) [inline src/main/java/com/mozilla/secops/authprofile/AuthProfile.java]: should this be auth_session?

Comment (ajvb) [reply src/main/java/com/mozilla/secops/authprofile/AuthProfile.java]: yes, thank you!

File: src/main/java/com/mozilla/secops/alert/Alert.java (hunk 2 above)
Comment (ajvb) [reply src/main/java/com/mozilla/secops/alert/Alert.java]: imo we can add that later if we need it. I imagine that we will only use templates for direct alerts and one template per notification system per pipeline for now.

File: src/main/java/com/mozilla/secops/GcsUtil.java (hunk 1 above)
Comment (ajvb) [reply src/main/java/com/mozilla/secops/GcsUtil.java]: Hmm, yeah probably. Would be worth logging if auth is failing.

**Comment by ajvb** [comment]
I believe something is up with travis ci. The error is related to a change that is definitely in this branch (`.getTemplateName()` being changed in `TestAuthProfile.java`).

**Pull request merged**